nautobot_golden_config/nornir_plays/config_compliance.py

```diff
@@ -64,6 +71,19 @@ def get_config_element(rule, config, obj, logger):
         else:
             config_element = config_json
 
+    elif rule["obj"].config_type == ComplianceRuleConfigTypeChoice.TYPE_XML:
+        config_xml = get_xml_config(config)
+
+        if not config_xml:
+            error_msg = "`E3002:` Unable to interpret configuration as XML."
+            logger.error(error_msg, extra={"object": obj})
+            raise NornirNautobotException(error_msg)
+
+        if rule["obj"].match_config:
+            config_element = get_xml_subtree_with_full_path(config_xml, rule["obj"].match_config)
```

Comment:
is there a chance a non-supported/incorrectly formatted Xpath could be provided here causing this to bomb out?

Reply:
Did some testing and I was able to get it throw exceptions using bad xpath,  added some error handling around it.

<img width="748" alt="Screenshot 2024-04-19 at 5 59 51 PM" src="https://github.com/nautobot/nautobot-app-golden-config/assets/47164813/bbfa34aa-bb16-49b7-9eb1-9962e37b3d7c">